### ISSUE TITLE
[Question] Could not resolve type with token

### ISSUE THREAD

**Issue by Bezarius** [protobuf-net/protobuf-net.Grpc#156]
Hey! 

I'm trying to use protobuf-net.Grpc in my .net 5.0 blazor wasm app, but it's doesn't work :\

From log:
```
info: ProtoBuf.Grpc.Server.ServicesExtensions.CodeFirstServiceMethodProvider[0]
      RPC services being provided by Portal.Infrastructure.Services.CalculatorService: 1
```

Judging by this log on the server side, everything is ok.

But when i'm trying to use service, i recieve error:

```
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: The type initializer for 'DefaultProxyCache`1' threw an exception.
System.TypeInitializationException: The type initializer for 'DefaultProxyCache`1' threw an exception.
 ---> System.TypeLoadException: Could not resolve type with token 0100001f from typeref (expected class 'System.ServiceModel.ServiceContractAttribute' in assembly 'System.ServiceModel.Primitives, Version=4.5.0.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a')
   at System.Reflection.CustomAttribute.GetCustomAttributesBase(ICustomAttributeProvider obj, Type attributeType, Boolean inheritedOnly)
   at System.Reflection.CustomAttribute.GetCustomAttributes(ICustomAttributeProvider obj, Type attributeType, Boolean inherit)
   at System.Attribute.GetAttr(ICustomAttributeProvider element, Type attributeType, Boolean inherit)
   at System.Attribute.GetCustomAttribute(MemberInfo element, Type attributeType)
   at System.Reflection.CustomAttributeExtensions.GetCustomAttribute(MemberInfo element, Type attributeType)
   at ProtoBuf.Grpc.Internal.ProxyEmitter.CreateFactory[ICalculatorService](BinderConfiguration binderConfig) in /_/src/protobuf-net.Grpc/Internal/ProxyEmitter.cs:line 119
   at ProtoBuf.Grpc.Configuration.ClientFactory.DefaultProxyCache`1[[Portal.Shared.ICalculatorService, Portal.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]..cctor() in /_/src/protobuf-net.Grpc/Configuration/ClientFactory.cs:line 81
   --- End of inner exception stack trace ---
   at ProtoBuf.Grpc.Configuration.ClientFactory.DefaultClientFactory.CreateClient[ICalculatorService](CallInvoker channel) in /_/src/protobuf-net.Grpc/Configuration/ClientFactory.cs:line 91
   at Portal.Client.Pages.Index.TestGrpc() in F:\DotNetProjects\Portal\Portal\Client\Pages\Index.razor.cs:line 24
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)
```

My code:

```
        protected async Task TestGrpc()
        {
            var httpClient = new HttpClient(new GrpcWebHandler(GrpcWebMode.GrpcWeb, new HttpClientHandler()));
            using (var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions { HttpClient = httpClient }))
            {
                var client = channel.CreateGrpcService<ICalculatorService>();
                _result = await client.MultiplyAsync(new MultiplyRequest { X = 12, Y = 4 });
                Console.WriteLine(_result);
            }
        }
```

I'm checked that example: https://github.com/mateusz-pietrzak/BlazorAppGrpcTest/ and it work with .net 5.0
(I pushed the project version up and updated dependencies.). 

I suspect there problem on client side, but i simplified everything to maximum, but nothing changed.

For test i moved code from BlazorAppGrpcTest into my app and it also doesn't work. Also i'm implemented simple grpc service using standart way using .proto and it work nice.

Please, tell me what am i doing wrong or what i should do to find issue? :) I spent 2 days but couldn't find the problem :\

**Comment by Bezarius** [comment]
The solution to the problem turned out to be extremely commonplace. It turns out I had to add System.ServiceModel.Primitives :\

**Comment by StevenTCramer** [comment]
@mgravell  I had to add the following:

<PackageReference Include="System.ServiceModel.Primitives" Version="4.8.1" />

But I don't understand why? It compiles and is indirectly added already. Thus is in the dependency chain.  Why does explicitly adding this to the package references fix the issue?

**Comment by mgravell** [comment]
Good question. Answers welcome!

**Comment by AshleyGrant** [comment]
I know this issue is several years old, but I am experiencing this exact issue today. I have added `System.ServiceModel.Primitives` to all of my projects, but the exception is still being thrown. 

I pulled down the code, added the `protobuf-net.Grpc` project to my own and started spelunking. It seems to be happening at a spot that makes absolutely no sense. I have been debugging into `ProxyEmitter.CreateFactory` and the crazy thing is I'm seeing exceptions thrown on code I've commented out.

I am at a loss, but I think the exception is being thrown when creating `BinderConfiguration.Default`. Is there any chance this issue can be reopened and looked at again?
